lib/postoffice/messaging.ex

```diff
@@ -48,21 +49,49 @@ defmodule Postoffice.Messaging do
   end
 
   @doc """
-  Creates a message.
+  Add a message to deliver marking this as pending.
 
   ## Examples
 
-      iex> create_message(%{field: value})
+      iex> add_message_to_deliver(%{field: value})
       {:ok, %Message{}}
 
-      iex> create_message(%{field: bad_value})
+      iex> add_message_to_deliver(%{field: bad_value})
       {:error, %Ecto.Changeset{}}
 
   """
-  def create_message(topic, attrs \\ %{}) do
-    Ecto.build_assoc(topic, :messages, attrs)
-    |> Message.changeset(attrs)
-    |> Repo.insert()
+  def add_message_to_deliver(topic, attrs \\ %{}) do
+    topic =
```

Comment:
actually, the only entry point to this method is through `receive_message`, and the topic should come already preloaded, but, as we are making a quick change and I want to modify as few things as possible (and that would change a lot of tests), I suggest to open an issue to remove this preload from here, and always expect a topic with everything loaded